securedrop/tests/test_unit_journalist.py

```diff
@@ -2,356 +2,393 @@
+        newTotp = self.user.totp
 
-        res = self.client.get(
-            url_for('admin_new_user_two_factor', uid=self.user.id)
-            )
+        # check that totp is different
+        self.assertNotEqual(oldTotp, newTotp)
 
-        # any GET req should take a user to the admin_new_user_two_factor page
-        self.assertIn('Authenticator', res.data)
+        # should redirect to verification page
+        self.assertRedirects(resp, url_for('account_new_two_factor'))
 
-    def test_admin_add_user_get_req(self):
+    def test_admin_resets_hotp_with_missing_otp_secret_key(self):
         self._login_admin()
+        resp = self.client.post(url_for('admin_reset_two_factor_hotp'),
+                                data=dict(uid=self.user.id))
 
-        res = self.client.get(url_for('admin_add_user'))
-
-        # any GET req should take a user to the admin_add_user page
-        self.assertIn('Add user', res.data)
+        self.assertIn('Change Secret', resp.data)
 
-    def test_admin_add_user_success(self):
+    def test_admin_new_user_2fa_redirect(self):
         self._login_admin()
+        resp = self.client.post(
+            url_for('admin_new_user_two_factor', uid=self.user.id),
+            data=dict(token='mocked'))
+        self.assertRedirects(resp, url_for('admin_index'))
 
-        res = self.client.post(
-            url_for('admin_add_user'),
-            data=dict(username='dellsberg',
-                      password='pentagonpapers',
-                      password_again='pentagonpapers',
-                      is_admin=False)
-            )
-
-        self.assertRedirects(res, url_for('admin_new_user_two_factor', uid=3))
-
-    def test_admin_add_user_failure_no_username(self):
+    def test_http_get_on_admin_new_user_two_factor_page(self):
         self._login_admin()
+        resp = self.client.get(url_for('admin_new_user_two_factor', uid=self.user.id))
+        # any GET req should take a user to the admin_new_user_two_factor page
+        self.assertIn('Authenticator', resp.data)
 
-        res = self.client.post(
-            url_for('admin_add_user'),
-            data=dict(username='', password='pentagonpapers',
-                      password_again='pentagonpapers', is_admin=False))
-
-        self.assertIn('Missing username', res.data)
-
-    def test_admin_add_user_failure_passwords_dont_match(self):
+    def test_http_get_on_admin_add_user_page(self):
         self._login_admin()
+        resp = self.client.get(url_for('admin_add_user'))
+        # any GET req should take a user to the admin_add_user page
+        self.assertIn('Add user', resp.data)
 
-        res = self.client.post(
-            url_for('admin_add_user'),
-            data=dict(username='dellsberg', password='not',
-                      password_again='thesame', is_admin=False))
-
-        self.assertIn('Passwords didn', res.data)
-
-    def test_admin_add_user_failure_password_too_long(self):
```

Comment:
No I renamed it `test_admin_edits_user_password_too_long_warning` since it doesn't test that the operation actually fails (i.e., that the database record is unmodified)--it only checks a warning is shown. Think it's important to be specific about that sort of thing.